cmd/snap-update-ns/utils_test.go

```diff
@@ -53,20 +61,67 @@ func (s *utilsSuite) TestSecureMkdirAllRelative(c *C) {
 	c.Assert(s.sys.Calls(), HasLen, 0)
 }
 
-// Ensure that we can create a directory with an absolute path.
-func (s *utilsSuite) TestSecureMkdirAllAbsolute(c *C) {
-	c.Assert(update.SecureMkdirAll("/abs/path", 0755, 123, 456), IsNil)
+// Ensure that we can "create the root directory.
+func (s *utilsSuite) TestSecureMkdirAllLevel0(c *C) {
+	c.Assert(update.SecureMkdirAll("/", 0755, 123, 456), IsNil)
 	c.Assert(s.sys.Calls(), DeepEquals, []string{
-		`open "/" O_NOFOLLOW|O_CLOEXEC|O_DIRECTORY 0`,
-		`mkdirat 3 "abs" 0755`,
-		`openat 3 "abs" O_NOFOLLOW|O_CLOEXEC|O_DIRECTORY 0`,
+		`open "/" O_NOFOLLOW|O_CLOEXEC|O_DIRECTORY 0`, // -> 3
+		`close 3`,
+	})
+}
+
+// Ensure that we can create a directory in the top-level directory.
+func (s *utilsSuite) TestSecureMkdirAllLevel1(c *C) {
```

Comment:
don't we need a test for for update.SecureMkDirAll("/") ? should that fail if perm or gid/uid aren't the only possible ones?

Reply:
I don't think this needs to be this paranoid. First of all, we don't chown things we don't create and we don't create / ever :-) (we just open it, there's no code to try to create it). Lastly we are still confined and we cannot open / in writable mode.

Reply:
either way we still need to test what happens

Reply:
I added a Level0 test here actually, let me see ...